src/button.test.jsx

```diff
@@ -63,73 +64,53 @@ describe('Button', () => {
 			label = 'Icon Button',
 			BUTTON_LABEL = 'button--label',
 			BUTTON_ICON = 'button--icon';
-
-		let buttonEl;
+		let button;
 
 		beforeEach(() => {
-			const button = TestUtils.renderIntoDocument(
+			button = TestUtils.renderIntoDocument(
 				<Button icon={icon} primary>
 					{label}
 				</Button>
 			);
-
-			buttonEl = ReactDOM.findDOMNode(button);
 		});
 
 		afterEach(() => {
-			buttonEl = null;
+			button = null;
 		});
 
-		it('creates a first-child icon element', () => {
-			const iconEl = buttonEl.firstChild;
-			expect(iconEl.classList.contains(BUTTON_ICON)).toBe(true);
+		it('should render a `Flex` container for icons and label', () => {
+			const flex = findComponentsWithType(button, 'Flex');
```

Comment:
testing for Flex/FlexItem is sort of okay, but doesn't seem to me like a very useful unit test since the `<Flex>` usage seems more like an implementation detail as opposed to something that is fundamental to a button being a button. Put another way, if the button was changed to use something other than `<Flex>`, should we require that tests be updated? I'm sort of ambivalent, but in general would just want there to be a test for the existence of an icon and a test for the existence of a label - those are baseline requirements for a button. Beyond that, you just have to make a call how much you want to 'set' the implementation requirements.

At a minimum, probably worth breaking out separate tests for `label` and `icon` independent of the component that wraps them.